src/irmin-pack/unix/sparse_file.ml

```diff
@@ -297,7 +293,7 @@ module Make (Io : Io.S) = struct
   let get_poff { mapping; _ } ~off =
     match Mapping_file.find_nearest_geq mapping off with
     | None ->
-        let s = Fmt.str "offset %a is after the prefix" Int63.pp off in
+        let s = Fmt.str "offset %a is after the sparse file" Int63.pp off in
```

Comment:
Since this error could refer to different files once we have volumes, it would be nice to include the path to either the mapping or data file in this error message.

Also, the error `Invalid_read_of_gced_object` might deserve a rename. Perhaps, `Invalid_read_of_sparse_object` (or something like this that is more generic).

Reply:
Yeah I agree that it would be nicer if all errors had the filenames for debugging user issues. Do you mind if we do a separate PR to improve the error reporting?
(I did switch the `Invalid_read_of_gced_object` for `Invalid_sparse_read` to match with the other existing errors)

Reply:
Yep, totally fine with a follow up PR to improve errors!